conan/tools/cmake/layout.py

```diff
@@ -22,15 +24,16 @@ def cmake_layout(conanfile, generator=None, src_folder=".", build_folder="build"
         raise ConanException("'build_type' setting not defined, it is necessary for cmake_layout()")
 
     build_folder = build_folder if not subproject else os.path.join(subproject, build_folder)
+    test_root_build_folder = get_root_test_build_folder(conanfile)  # only for test_package's
     config_build_folder, user_defined_build = get_build_folder_custom_vars(conanfile)
+
     if config_build_folder:
-        build_folder = os.path.join(build_folder, config_build_folder)
+        build_folder = os.path.join(test_root_build_folder, build_folder, config_build_folder)
```

Comment:
There is something strange here. This is not only for test, but it also applies to normal developer layouts, where the "build" folder is. So there should be a conditional: "if we are in a test_package, use this ``test_root_build_folder``, or use ``build_folder`` otherwise"

Reply:
If it's a test folder, it will be an empty string (the conditional is inside the function), so there is no need to put more conditionals there.

Reply:
Yes, but I am afraid the expected order is the reversed one, the test-folder is closer (or substitute for the ``config_build_folder``.